From ac79a51a87dc9e7729483786c24b7c04f7ab7ddd Mon Sep 17 00:00:00 2001
Message-ID: <ac79a51a87dc9e7729483786c24b7c04f7ab7ddd.1747779548.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1747779547.git.sharlatanus@gmail.com>
References: <cover.1747779547.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 19 May 2025 19:41:50 +0100
Subject: [PATCH 22/31] gnu: python-healpy: Update to 1.18.1.

* gnu/packages/astronomy.scm (python-healpy): Update to 1.18.1.
[arguments] <test-flags>: Provide "--pyargs" option to test against
installed module.
<phases>: Remove 'disable-doctestplus, and 'pre-check.
[native-inputs]: Add python-pytest-doctestplus.

Change-Id: I4f32f7ca551d556782a545171da30f6acc6e1c2a
---
 gnu/packages/astronomy.scm | 38 +++++++++++++++-----------------------
 1 file changed, 15 insertions(+), 23 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index e45c104935..6c13621880 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -3882,33 +3882,25 @@ (define-public python-hasasia
 (define-public python-healpy
   (package
     (name "python-healpy")
-    (version "1.18.0")
+    (version "1.18.1")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "healpy" version))
        (sha256
-        (base32 "12ajn55kjgnqadyamh6cw4q8i01nyv6isgf47lcnv2jch27zs4ka"))))
+        (base32 "1v7bsxz05k36cw596yyaahcx6h8blhy1qlzpwcjj5cvg4vrynnff"))))
     (build-system pyproject-build-system)
     (arguments
      (list
       #:test-flags
-      ;; Disable tests requiring network access.
-      #~(list "-k" (string-append "not test_astropy_download_file"
-                                  " and not test_pixelweights_local_datapath"
-                                  " and not test_rotate_map_polarization_alms"))
-      #:phases
-      #~(modify-phases %standard-phases
-          ;; XXX: It's not compatible with pytest-8, enable when newer version
-          ;; is available.
-          (add-after 'unpack 'disable-doctest
-            (lambda _
-            (substitute* "pyproject.toml"
-              (("--doctest-plus") ""))))
-          (add-before 'check 'pre-check
-            (lambda _
-              (delete-file "lib/healpy/conftest.py")
-              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+      #~(list "--pyargs" "healpy"
+              "-k" (string-join
+                    ;; Tests requiring network access.
+                    (list "not test_astropy_download_file"
+                          "test_pixelweights_local_datapath"
+                          "test_rotate_map_polarization_alms")
+                    " and not ")
+              "test")))
     (native-inputs
      (list nss-certs-for-test
            pkg-config
@@ -3916,20 +3908,20 @@ (define-public python-healpy
            python-pytest
            python-pytest-astropy-header
            python-pytest-cython
-           ;python-pytest-doctestplus
+           python-pytest-doctestplus
            python-setuptools
            python-setuptools-scm
            python-wheel))
+    (inputs
+     (list cfitsio
+           healpix-cxx
+           libsharp))
     (propagated-inputs
      (list python-astropy
            python-colorlog
            python-matplotlib
            python-numpy
            python-scipy))
-    (inputs
-     (list cfitsio
-           healpix-cxx
-           libsharp))
     (home-page "http://healpy.readthedocs.org/")
     (synopsis "Healpix tools package for Python")
     (description
-- 
2.49.0

